src/ring_theory/algebra.lean

```diff
@@ -26,55 +26,37 @@ ring is the under category R ↓ CRing. In the categorical
 setting we have a forgetful functor R-Alg ⥤ R-Mod.
 However here it extends module in order to preserve
 definitional equality in certain cases. -/
-class algebra (R : Type u) (A : Type v) [comm_ring R] [ring A] extends has_scalar R A :=
-(to_fun : R → A) [hom : is_ring_hom to_fun]
+@[nolint has_inhabited_instance]
+class algebra (R : Type u) (A : Type v) [comm_semiring R] [semiring A] extends has_scalar R A :=
```

Comment:
Is there a reason that you don't extend `ring_hom`, but take the ring hom as a field instead?

Reply:
`ring_hom` is not a `class` so I'm not sure that it will play nice with `extends`.